src/sparsebase/io/pigo_mtx_reader.cc

```diff
@@ -23,12 +36,117 @@ PigoMTXReader<IDType, NNZType, ValueType>::PigoMTXReader(
 //  return reader.ReadArray();
 //}
 
+template <typename IDType, typename NNZType, typename ValueType>
```

Comment:
I think it would make more sense if both MTXReader and PigoMTXReader shared the same enums and ParseHeader function, right? Is there a difference between the two? And if not, can you make these MTX definitions shared between them? It minimizes code duplication and makes it more maintainable.